Assets/Scripts/Algorithms/Patrolling/Components/Redistribution/DistanceRedistributionComponent.cs

```diff
@@ -0,0 +1,124 @@
+// Copyright 2025 MAEPS
+// 
+// This file is part of MAEPS
+// 
+// MAEPS is free software: you can redistribute it and/or modify it under
+// the terms of the GNU General Public License as published by the
+// Free Software Foundation, either version 3 of the License, or (at your option)
+// any later version.
+// 
+// MAEPS is distributed in the hope that it will be useful, but WITHOUT
+// ANY WARRANTY; without even the implied warranty of MERCHANTABILITY
+// or FITNESS FOR A PARTICULAR PURPOSE. See the GNU General
+// Public License for more details.
+// 
+// You should have received a copy of the GNU General Public License along
+// with MAEPS. If not, see http://www.gnu.org/licenses/.
+//
+// Contributors 2025: 
+// Casper Nyvang Sørensen,
+// Christian Ziegler Sejersen,
+// Jakob Meyer Olsen
+// Mads Beyer Mogensen
+using System;
+using System.Collections.Generic;
+using System.Linq;
+
+using Maes.Map;
+using Maes.Robot;
+
+namespace Maes.Algorithms.Patrolling.Components.Redistribution
+{
+    public sealed class DistanceRedistributionComponent : IComponent
+    {
+        private readonly IRobotController _controller;
+        private readonly IReadOnlyList<Vertex> _vertices;
+        private readonly int _probabilityFactor;
+        private readonly int _delay;
+        private readonly IReadOnlyDictionary<(int, int), float> _probabilityForPartitionSwitch;
+
+        public int PreUpdateOrder => -251;
+
+        public int PostUpdateOrder => -251;
+
+        public DistanceRedistributionComponent(IRobotController controller, IReadOnlyList<Vertex> vertices, int probabilityFactor, int delay = 1)
+        {
+            _controller = controller;
+            _vertices = vertices;
+            _probabilityFactor = probabilityFactor;
+            _delay = delay;
+            _probabilityForPartitionSwitch = CalculateClosestDistanceToPartitions(vertices);
+        }
+
+        public IEnumerable<ComponentWaitForCondition> PreUpdateLogic()
+        {
+            var random = new Random();
```

Comment:
You should **always** seed your random number generators. Otherwise using this component will make every run different.